core/controlplane/config/templates/stack-template.json

```diff
@@ -257,15 +242,37 @@
                   "Resource": "*",
                   "Effect": "Allow"
                 }
+          ]
+        }
+      }
+    },
+    "IAMRoleController": {
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": [
+                "sts:AssumeRole"
               ],
-              "Version": "2012-10-17"
-            },
-            "PolicyName": "root"
-          }
-        ]
+              "Effect": "Allow",
+              "Principal": {
+                "Service": [
+                  "ec2.{{.Region.PublicDomainName}}"
+                ]
+              }
+            }
+          ],
+          "Version": "2012-10-17"
+        },
+        "Path": "/",
+        {{if .Controller.IAMConfig.ManagedIamRoleName }}
+        "RoleName":  {"Fn::Join": ["",[{"Ref": "AWS::Region"},"-","{{.Controller.IAMConfig.ManagedIamRoleName}}"]]},
+        {{end}}
+        "ManagedPolicyArns": [ {"Ref": "IAMManagedPolicyController"} ]
```

Comment:
@Fsero Thanks for the detailed reply :bow:

> i like the first slightly more because the option is more clear either you go with a role or either you set an instance profile but you cannot do both.

It does make sense! Now, I like the first more, too. Let's go ahead with it 👍